create.go

```diff
@@ -284,3 +292,58 @@ func createPIDFile(pidFilePath string, pid int) error {
 
 	return nil
 }
+
+// copyIfNeeded copies the cpuset.cpus and cpuset.mems from the parent
+// directory to the current directory if the file's contents are 0
+func copyIfNeeded(current, parent string) error {
+	var (
+		err                      error
+		currentCpus, currentMems []byte
+		parentCpus, parentMems   []byte
+	)
+
+	if currentCpus, currentMems, err = getSubsystemSettings(current); err != nil {
+		return err
+	}
+	if parentCpus, parentMems, err = getSubsystemSettings(parent); err != nil {
+		return err
+	}
+
+	if isEmpty(currentCpus) {
+		if err := writeFile(current, "cpuset.cpus", string(parentCpus)); err != nil {
+			return err
+		}
+	}
+	if isEmpty(currentMems) {
+		if err := writeFile(current, "cpuset.mems", string(parentMems)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getSubsystemSettings(parent string) (cpus []byte, mems []byte, err error) {
+	if cpus, err = ioutil.ReadFile(filepath.Join(parent, "cpuset.cpus")); err != nil {
+		return
+	}
+	if mems, err = ioutil.ReadFile(filepath.Join(parent, "cpuset.mems")); err != nil {
+		return
+	}
+	return cpus, mems, nil
+}
+
+func isEmpty(b []byte) bool {
+	return len(bytes.Trim(b, "\n")) == 0
+}
+
+func writeFile(dir, file, data string) error {
```

Comment:
This is a relatively general-purpose routine so I would either move it to `utils.go` and change it to be:

```
func writefile(filePath, data string) error
```

.. or make maybe make it a cgroup-specific function by changing its name.

Reply:
I wonder if  `isEmpty` should be moved to util.go
```
func isEmpty(b []byte) bool
```

Reply:
@Ace-Tang Yes, good point. Let's move it there, and also rename it to `isEmptyString()`.